### ISSUE TITLE
[JavaScript] Missing model annotations in generated JSDoc?

### ISSUE THREAD

**Issue by delenius** [swagger-api/swagger-codegen#3164]
When creating a JS client with the `emitJSDoc=true` option, the parameters of the model classes are lacking type annotations. For example, I have a generated `model/Document.js` class with this constructor:

```javascript
  /**
   * Constructs a new <code>Document</code>.
   * @alias module:model/Document
   * @class
   * @param documentId
   * @param modified
   */
  var exports = function(documentId, modified) {
    ...
```

The generated HTML page for this model class has a table, which is also missing this type information, as seen here:
![screenshot 2016-06-17 16 05 08](https://cloud.githubusercontent.com/assets/7936653/16167108/308c6480-34a6-11e6-908b-5659e8e1b7cc.png)

Somehow the listings below the table *does* have the information. Not sure where it got it.

Also, how can I create annotations in Java that will generate entries in the "Description" fields in the table?

The generated markdown version looks like this:

```markdown
# Sunflower.Document

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**documentId** | **String** |  | 
**modified** | **Boolean** |  | [default to false]
```

So, I'm not sure where the problem needs to be addressed. But shouldn't the JavaScript look like this:

```javascript
  /**
   * Constructs a new <code>Document</code>.
   * @alias module:model/Document
   * @class
   * @param documentId {String}
   * @param modified {Boolean}
   */
  var exports = function(documentId, modified) {
    ...
```

**Comment by wing328** [comment]
@delenius  I found the following syntax:
```
/**
 * Repeat <tt>str</tt> several times.
 * @param {string} str The string to repeat.
 * @param {number} [times=1] How many times to repeat the string.
 * @returns {string}
 */
```
Ref: http://speakingjs.com/es5/ch29.html

so looks like the parameter type should be in curly brackets right after @param

**Comment by delenius** [comment]
@wing328 right, so that needs to be added to the JS generator.

**Comment by wing328** [comment]
@delenius may I know if you've cycle to contribute the enhancement?

**Comment by delenius** [comment]
@wing328 ok, PR is in.

Note that some of the JS integration-tests fail, but it looks like they haven't been updated for a while, and the failure looks unrelated to my changes.

The bin/javascript-petstore.sh script even generates new files that are not tracked.

It would be good if these generated test files weren't actually tracked by the VCS, but rather generated on the fly by the CI system to do the integration testing. Most people seem unaware that you're supposed to go through these steps, so these files are almost always out of sync.

**Comment by wing328** [comment]
@delenius the issue is due to new test case in `modules/swagger-codegen/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml` to cover nested array/map of enum.

I'll take a look later today.

Yes, I agree with you that ideally the test should generate the updated Petstore API client but the problem is that there's no easy way to tell which language has been updated in the PR and updating Petstore API client for **all** langauges is not a good idea as not every generator can handle the new test cases we added to `petstore-with-fake-endpoints-models-for-testing.yaml`

It's also a good practice to update the local Petstore API client to test the change locally first before submitting the PR.
